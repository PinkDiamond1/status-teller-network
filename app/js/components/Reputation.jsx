import React from 'react';
import PropTypes from 'prop-types';
import {withNamespaces} from 'react-i18next';
import RatingIcon from "./RatingIcon";
import classnames from "classnames";

import './Reputation.scss';

const Reputation = ({reputation, size}) => (
  <span className={classnames("reputation-container", {small: size === 's', large: size === 'l'})}>
    <span className="left-rating bg-secondary">
      {reputation.upCount} <RatingIcon isPositiveRating={true}/>
    </span>
    <span className="right-rating bg-secondary">
      {reputation.downCount} <RatingIcon isPositiveRating={false}/>
    </span>
  </span>
);

Reputation.propTypes = {
  t: PropTypes.func,
  reputation: PropTypes.object,
  size: PropTypes.string

};

export default withNamespaces()(Reputation);
