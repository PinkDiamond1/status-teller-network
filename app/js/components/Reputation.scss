.reputation-container {
  .left-rating, .right-rating {
    border-top-right-radius: 32px;
    border-bottom-right-radius: 32px;
    font-size: 15px;
    margin: 0 1px;
    width: 70px;
    height: 40px;
    display: inline-block;
    line-height: 40px;
    text-align: center;
  }

  .left-rating {
    border-radius: 32px 0 0 32px;
  }

  &.small {
    .left-rating, .right-rating {
      font-size: 13px;
      width: 55px;
      height: 24px;
      line-height: 22px;
      border-top-right-radius: 15px;
      border-bottom-right-radius: 15px;
    }

    &.left-rating {
      border-radius: 15px 0 0 15px;
    }
  }

}
