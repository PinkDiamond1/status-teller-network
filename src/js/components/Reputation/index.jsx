/* eslint-disable no-confusing-arrow */
import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {withNamespaces} from 'react-i18next';
import classnames from "classnames";

import RatingIcon from "../../ui/RatingIcon";

import './index.scss';

class Reputation extends Component {
  rateTrade(rating) {
    if (!this.clickable) {
      return;
    }
    this.props.rateTransaction(this.props.trade.escrowId, rating);
  }

  render() {
    const {size, rateTransaction, trade, reputation = {}} = this.props;
    const rating = trade ? parseInt(trade.rating, 10) : 0;
    const tradeWasRated = trade && rating !== 0;
    this.clickable = !tradeWasRated && rateTransaction && trade;

    if (reputation.hasOwnProperty('averageCount')) {
      let classes = 'average-rating py-1 px-3 rounded ';
      if (!Number.isInteger(reputation.averageCount)) {
        classes += 'bg-dark text-dark';
      } else  if (reputation.averageCount >= 4) {
        classes += 'bg-success text-success';
      } else  if (reputation.averageCount < 4 && reputation.averageCount > 2) {
        classes += 'bg-warning text-warning';
      } else  if (reputation.averageCount < 2) {
        classes += 'bg-danger text-danger';
      }

      return <span className={classes}>
        {Number.isInteger(reputation.averageCount) ? reputation.averageCount : 'NA'}
      </span>;
    }

    return <span className={classnames("reputation-container", {small: size === 's', large: size === 'l'})}>
      <span className={classnames("left-rating", {
        "bg-primary": tradeWasRated && rating === 5,
        clickable: this.clickable
      })}>
      {(tradeWasRated && reputation.upCount) || (!trade && reputation.upCount)}
        &nbsp;<RatingIcon isPositiveRating={true} onClick={() => this.rateTrade('5')}/>
      </span>
      <span className={classnames("right-rating", {
        "bg-primary": tradeWasRated && rating === 1,
        clickable: this.clickable
      })}>
      {(tradeWasRated && reputation.downCount) || (!trade && reputation.downCount)}
        &nbsp;<RatingIcon isPositiveRating={false} onClick={() => this.rateTrade('1')}/>
      </span>
    </span>;
  }
}

Reputation.propTypes = {
  t: PropTypes.func,
  size: PropTypes.string,
  trade: PropTypes.object,
  rateTransaction: PropTypes.func,
  reputation: PropTypes.object
};

export default withNamespaces()(Reputation);
