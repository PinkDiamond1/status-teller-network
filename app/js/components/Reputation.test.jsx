import React from "react";
import { shallow } from "enzyme";

import Reputation from "./Reputation";

describe('Trades', () => {
  it('should render correctly', () => {
    const component = shallow(<Reputation />);
  
    expect(component).toMatchSnapshot();
  });
});
