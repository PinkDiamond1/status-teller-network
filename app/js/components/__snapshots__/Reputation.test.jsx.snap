// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trades should render correctly 1`] = `
<span
  className="reputation-container"
>
  <span
    className="left-rating bg-secondary"
  >
    1
     
    <RatingIcon
      isPositiveRating={true}
    />
  </span>
  <span
    className="right-rating bg-secondary"
  >
    2
     
    <RatingIcon
      isPositiveRating={false}
    />
  </span>
</span>
`;
